pkg/sharding/ranges.go

```diff
@@ -82,6 +83,10 @@ func logRangesFromPath(path string) (Ranges, error) {
 		return Ranges{}, nil
 	}
 	if err := yaml.Unmarshal(contents, &ranges); err != nil {
+		// Try to use JSON
+		if jerr := json.Unmarshal(contents, &ranges); jerr == nil {
+			return ranges, nil
```

Comment:
If we don't assign this jerr to err, there can be a situation where we have a nil error, but a zero-value range.